src/labeler.ts

```diff
@@ -214,19 +261,31 @@ function checkAll(changedFiles: string[], globs: string[]): boolean {
 }
 
 function checkMatch(changedFiles: string[], matchConfig: MatchConfig): boolean {
-  if (matchConfig.all !== undefined) {
-    if (!checkAll(changedFiles, matchConfig.all)) {
-      return false;
+  if (matchConfig.changedFiles?.all !== undefined) {
+    if (checkAll(changedFiles, matchConfig.changedFiles.all)) {
+      return true;
     }
   }
 
-  if (matchConfig.any !== undefined) {
-    if (!checkAny(changedFiles, matchConfig.any)) {
-      return false;
+  if (matchConfig.changedFiles?.any !== undefined) {
+    if (checkAny(changedFiles, matchConfig.changedFiles.any)) {
+      return true;
     }
   }
 
-  return true;
+  if (matchConfig.headBranch !== undefined) {
+    if (checkBranch(matchConfig.headBranch, 'head')) {
+      return true;
+    }
+  }
```

Comment:
Could you please change it to:
```ts
  if (matchConfig.headBranch) {
    return checkBranch(matchConfig.headBranch, 'head');
  }
```
and the same for others.

Reply:
I was just following the existing pattern that was here, but I'm more than happy to update that as I prefer that syntax personally 👍

Reply:
Does this break if both a head branch and a base branch are present? In that case, I think we would need to do both checks